receiver/otlpreceiver/otlp.go

```diff
@@ -88,12 +88,13 @@ func newOtlpReceiver(cfg *Config, set *receiver.Settings) (*otlpReceiver, error)
 
 func (r *otlpReceiver) startGRPCServer(host component.Host) error {
 	// If GRPC is not enabled, nothing to start.
-	if r.cfg.GRPC == nil {
+	if !r.cfg.GRPC.HasValue() {
 		return nil
 	}
 
+	grpcCfg := r.cfg.GRPC.Get()
```

Comment:
We could simplify these cases by extending the `Get()` to return (T, bool) instead.

Reply:
What about creating another function for pointer return `ToPointer` and `Get` returning the value? The return type could be a private empty one `return empty[T](), false`

Reply:
I was thinking about this during the PR for adding the configoptional package. I like that `HasValue` gates the rest of the function and avoids nesting, but I see the value in the other approaches as well.

My thought was that we see how this is used in a few places and see which pattern ends up being the cleanest for the typical case. From what I can tell, outside the reduced API of removing `HasValue` and checking whether `*T` is nil or  `ok` is true for `(T, bool)`, none of the options are substantially cleaner than the others.